core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1009,26 +910,13 @@ private FileIO tableFileIO(SessionContext context, Map<String, String> config) {
     return newFileIO(context, fullConf);
   }
 
-  private AuthSession tableSession(Map<String, String> tableConf, AuthSession parent) {
-    Map<String, String> credentials = Maps.newHashMapWithExpectedSize(tableConf.size());
-    for (String prop : tableConf.keySet()) {
-      if (TABLE_SESSION_ALLOW_LIST.contains(prop)) {
-        credentials.put(prop, tableConf.get(prop));
-      }
-    }
-
-    Pair<String, Supplier<AuthSession>> newSession = newSession(credentials, tableConf, parent);
-    if (null == newSession) {
-      return parent;
-    }
-
-    AuthSession session = tableSessions.get(newSession.first(), id -> newSession.second().get());
-
-    return session != null ? session : parent;
+  private AuthSession tableSession(
+      TableIdentifier id, Map<String, String> tableConf, AuthSession parent) {
+    return authManager.tableSession(id, tableConf, parent);
   }
 
   private static ConfigResponse fetchConfig(
-      RESTClient client, Map<String, String> headers, Map<String, String> properties) {
+      RESTClient client, AuthSession initialAuth, Map<String, String> properties) {
```

Comment:
I'm concerned that we're replacing headers with session.  There are headers independent of the auth session.

Reply:
Fixed. `fetchConfig` is now using `configHeaders`, see fbc3555.